analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.LearnConstraint.cs

```diff
@@ -470,8 +338,20 @@ public void Branching_LearnsObjectConstraint_TypePattern(string expression)
         [DataRow("arg is not object")]
         [DataRow("arg is not Exception")]
         public void Branching_LearnsObjectConstraint_TypePattern_Negated(string expression)
+        {
+            var validator = CreateIfElseEndValidatorCS(expression, OperationKind.TypePattern);
+            validator.ValidateTag("If", x => x.Should().BeNull("it could be null or any other type"));
+            validator.ValidateTag("Else", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.TagValues("End").Should().HaveCount(2)
+                .And.ContainSingle(x => x == null)
+                .And.ContainSingle(x => x != null && x.HasConstraint(ObjectConstraint.NotNull));
+        }
+
+        private static ValidatorTestCheck CreateIfElseEndValidatorCS(string expression, OperationKind expectedOperation)
         {
             var code = @$"
+object isNull = null;
+var isObject = new object();
 if ({expression})
 {{
```

Comment:
Could be written shorter by using `_ = {expression} ? Tag() : Tag();`

Reply:
It is intentionally explicit to express the execution flow. I could remove the `{{` and `}}` to make it shorter if you want.

Reply:
> I could remove the {{ and }} to make it shorter if you want.

I wouldn't do that. It is more flexible and in line with our coding style.